### PR TITLE
docs(azure): update example and readme to use Entra ID

### PR DESCRIPTION
**Pull request by deyaaeldeen**
- [x] I understand that this repository is auto-generated and my pull request may not be merged

## Changes being requested

Update Azure code examples to use Entra ID instead of API keys.

### CODE REVIEW

File: examples/azure.ts [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 // Corresponds to your Model deployment within your OpenAI resource, e.g. gpt-4-1106-preview
 // Navigate to the Azure OpenAI Studio to deploy a model.
 const deployment = 'gpt-4-1106-preview';
 
+const credential = new DefaultAzureCredential();
+const scope = 'https://cognitiveservices.azure.com/.default';
+const azureADTokenProvider = getBearerTokenProvider(credential, scope);
+
 // Make sure to set both AZURE_OPENAI_ENDPOINT with the endpoint of your Azure resource and AZURE_OPENAI_API_KEY with the API key.
 // You can find both information in the Azure Portal.
```
Comment (RobertCraigie) [inline examples/azure.ts:15]: nit: this comment is outdated now right? I think we should list out the different ways to authenticate here

File: examples/azure.ts [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 import { AzureOpenAI } from 'openai';
+import { getBearerTokenProvider, DefaultAzureCredential } from '@azure/identity';
 
 // Corresponds to your Model deployment within your OpenAI resource, e.g. gpt-4-1106-preview
 // Navigate to the Azure OpenAI Studio to deploy a model.
 const deployment = 'gpt-4-1106-preview';
 
+const credential = new DefaultAzureCredential();
+const scope = 'https://cognitiveservices.azure.com/.default';
+const azureADTokenProvider = getBearerTokenProvider(credential, scope);
```
Comment (RobertCraigie) [inline examples/azure.ts:12]: question: is this the most common authentication pattern for Azure OpenAI users?

Comment (deyaaeldeen) [reply examples/azure.ts:12]: Yes, it is Microsoft's enterprise authentication solution.

File: examples/azure.ts (hunk 1 above)
Comment (deyaaeldeen) [reply examples/azure.ts]: Good catch, addressed in https://github.com/openai/openai-node/pull/857/commits/32ee4eb3f40aece122713d68c29fb2e5cce2b970.

File: README.md [hunk 3]
```diff
@@ -367,11 +367,18 @@ To use this library with [Azure OpenAI](https://learn.microsoft.com/en-us/azure/
 class instead of the `OpenAI` class.
 
 > [!IMPORTANT]
-> The Azure API shape differs from the core API shape which means that the static types for responses / params
+> The Azure API shape slightly differs from the core API shape which means that the static types for responses / params
 > won't always be correct.
 
 ```ts
-const openai = new AzureOpenAI();
+import { AzureOpenAI } from 'openai';
```
Comment (RobertCraigie) [inline README.md:374]: I'm realising now that we don't mention the environment variables the azure client supports, e.g. `AZURE_OPENAI_ENDPOINT` & `AZURE_OPENAI_API_KEY`, we should mention these in the README as well. (doesn't block this PR)

**Review by RobertCraigie — APPROVED** [review]
Thank you!

I think CI should run if you target the `next` branch fyi

**Comment by deyaaeldeen** [comment]
> I think CI should run if you target the next branch fyi

Done.

**Pull request merged**